feat(stream): support exclusive range for XPENDING command

i18nsite opened the pull request:
[English](#en) / [中文](#cn)

<a id="en"></a>
### English

**Issue**: https://github.com/apache/kvrocks/issues/3276

**Description**:
This PR adds support for exclusive ranges in the `XPENDING` command, allowing users to specify exclusive start and end IDs using the `(` prefix, similar to `XRANGE` and `XREVRANGE`.

- Modified `StreamPendingOptions` to include `exclude_start` and `exclude_end` flags.
- Updated `CommandXPending::Parse` to detect the `(` prefix in `start_id` and `end_id` arguments.
- Updated `Stream::GetPendingEntries` to correctly filter entries based on the exclusion flags.
- Added comprehensive Go tests to verify the exclusive range functionality.

<a id="cn"></a>
### 中文

**Issue**: https://github.com/apache/kvrocks/issues/3276

**描述**:
本 PR 为 `XPENDING` 命令添加了对排他范围的支持，允许用户使用 `(` 前缀指定排他的起始和结束 ID，类似于 `XRANGE` 和 `XREVRANGE`。

- 修改了 `StreamPendingOptions`，增加了 `exclude_start` 和 `exclude_end` 标志。
- 更新了 `CommandXPending::Parse`，以检测 `start_id` 和 `end_id` 参数中的 `(` 前缀。
- 更新了 `Stream::GetPendingEntries`，根据排他标志正确过滤条目。
- 添加了全面的 Go 测试以验证排他范围功能。

i18nsite committed:
fix(stream): use correct end_id in XPENDING command (#3276)

The XPENDING command was incorrectly using the start_id instead of the end_id
when parsing the range arguments. This caused incorrect filtering of pending
entries when a range was specified.

This commit fixes the issue by passing the correct end_id to the
ParseStreamEntryID function.

PragmaTwice commented on tests/gocase/unit/type/stream/xpending_test.go:
```diff
@@ -0,0 +1,197 @@
+	"github.com/stretchr/testify/require"
+)
+
+func TestXPendingBug(t *testing.T) {
```
Is this function related to this PR?

i18nsite committed:
feat(stream): support exclusive range for XPENDING command (#3276)

This commit adds support for exclusive ranges in the XPENDING command
using the  prefix for start and end IDs, similar to XRANGE.

- Modified  to include  and  flags.
- Updated  to handle  prefix.
- Updated  to filter entries based on exclusion flags.
- Added Go tests to verify exclusive range behavior.

aleksraiden committed:
Merge branch 'unstable' into xpending_exclusive

i18nsite committed:
Merge branch 'unstable' into xpending_exclusive

aleksraiden committed:
Merge branch 'unstable' into xpending_exclusive

jihuayu commented:
@i18nsite Hi! If you want to keep this PR moving forward, could you please:

1. Resolve the merge conflicts  
2. Make sure CI passes

Thanks!

jihuayu commented:
There has been no activity on this PR; closing it.
